node/src/components/storage.rs

```diff
@@ -1616,86 +1579,194 @@ fn initialize_block_body_v1_db(
             }
         }
     }
+
     txn.commit()?;
     info!("v1 block body database initialized");
     Ok(())
 }
 
-fn delete_merkle_block_body_part(
-    tx: &mut RwTransaction,
-    block_body_v2_db: &Database,
-    value_database: &Database,
-    key: Digest,
-) -> Result<Digest, LmdbExtError> {
-    let [key_to_part_db, merkle_proof_of_rest]: [Digest; 2] =
-        match tx.get_value(*block_body_v2_db, &key)? {
+fn get_merkle_linked_list_node<Tx, T>(
+    tx: &mut Tx,
+    block_body_v2_db: Database,
+    part_database: Database,
+    key_to_block_body_db: &Digest,
+) -> Result<Option<MerkleLinkedListNode<T>>, LmdbExtError>
+where
+    Tx: Transaction,
+    T: DeserializeOwned,
+{
+    let [part_to_value_db, merkle_proof_of_rest]: [Digest; 2] =
```

Comment:
Should this rather be `merkle_root_of_rest`?

Reply:
The difference being that Merkle proof usually means the whole tree of hashes? In such a case, you're right, as it's only a single, top-level hash.

Reply:
Right, maybe `merkle_node_hash`? Anyway, I'm fine with "proof", too; just a thought.